ptero_workflow/api/v1/views.py

```diff
@@ -44,6 +45,26 @@ def get(self, workflow_id):
             LOG.exception('Unexpected exception getting workflow')
             raise
 
+class ExecutionInputsView(Resource):
+    def get(self, execution_id):
+        try:
+            execution_input_data = g.backend.get_execution_inputs(execution_id)
+            return execution_input_data, 200
+        except:
+            LOG.exception('Unexpected exception getting execution inputs')
+            raise
+
+class ExecutionOutputsView(Resource):
+    def put(self, execution_id):
+        try:
+            g.backend.set_execution_outputs(execution_id, request.get_json())
+            return '', 200
+        except OutputsAlreadySet:
+            return '', 403
```

Comment:
Hmm..I agree it might not be perfect use of 409, but 403 doesn't seem right to me.

If we make setting outputs more idempotent, then in theory the user could repeat the query by fixing their request to match the pre-exisitng outputs (though that would be silly to actually do that).  I think that 409 would be a reasonable status code for mismatched outputs if we move to the idempotent model.